src/ext/java/org/opentripplanner/ext/legacygraphqlapi/IntrospectionTypeWiring.java

```diff
@@ -18,9 +22,28 @@ class IntrospectionTypeWiring {
         Arrays.asList(method.getReturnType().getInterfaces()).contains(DataFetcher.class)
       );
 
-  static <T> TypeRuntimeWiring build(Class<T> clazz) throws Exception {
+  private final TypeDefinitionRegistry typeRegistry;
+
+  public IntrospectionTypeWiring(TypeDefinitionRegistry typeRegistry) {
+    this.typeRegistry = typeRegistry;
+  }
+
+  <T> TypeRuntimeWiring build(Class<T> clazz) throws Exception {
     T instance = clazz.getConstructor().newInstance();
 
+    String typeName = clazz
+      .getSimpleName()
+      .replaceFirst("LegacyGraphQL", "")
+      .replaceAll("Impl$", "");
+
+    TypeDefinition type = typeRegistry
+      .getType(typeName)
+      .orElseThrow(() -> new IllegalArgumentException("Type not found in schema"));
+
+    if (!(type instanceof ObjectTypeDefinition objectType)) {
+      throw new IllegalArgumentException("Type is not object type");
+    }
```

Comment:
This just adds some validation that is based on the current implementation class  naming convention, right? Should we include the `typename` in the exceptions so it would make it easier to understand what went wrong?

Reply:
Done